connaisseur/validators/cosign/cosign_validator.py

```diff
@@ -48,36 +49,43 @@ def __init__(
-        threshold = kwargs.get(
-            "threshold",
-            1 if trust_root != "*" or any(required) else len(self.trust_roots),
-        )
-        # vals is a validations dict for each required trust root containing validated
-        # digests and errors
-        vals = self.__get_pinned_trust_roots(trust_root, required, threshold)
-
-        # use concurrent.futures for now
-        # tasks = [self.__validation_task(k, str(image)) for k in self.vals.keys()]
-        # await asyncio.gather(*tasks)
-
-        # prepare executor
-        num_workers = len(vals)
-        executor = ThreadPoolExecutor(num_workers)
-        # prepare tasks
-        # a copy of vals dictionaries is passed to concurrent validation to ensure
-        # thread-safe execution
-        arguments = [(k, v.copy(), str(image)) for k, v in vals.items()]
-        futures = [executor.submit(self.__validation_task, *arg) for arg in arguments]
-        # await results (output dropped as `vals` is updated within function)
-        for future in futures:
-            vals.update(future.result())
-
-        return CosignValidator.__apply_policy(
-            vals=vals, threshold=threshold, required=required
-        )
+        try:
+            required = kwargs.get("required", [])
+            # if not configured, `threshold` is 1 if trust root is not "*" or
+            # `required` is specified and number of trust roots otherwise
+            threshold = kwargs.get(
+                "threshold",
+                1 if trust_root != "*" or any(required) else len(self.trust_roots),
+            )
+            # vals is a validations dict for each required trust root containing validated
+            # digests and errors
+            vals = self.__get_pinned_trust_roots(trust_root, required, threshold)
+
+            # use concurrent.futures for now
+            # tasks = [self.__validation_task(k, str(image)) for k in self.vals.keys()]
+            # await asyncio.gather(*tasks)
+
+            # prepare executor
+            num_workers = len(vals)
+            executor = ThreadPoolExecutor(num_workers)
+            # prepare tasks
+            # a copy of vals dictionaries is passed to concurrent validation to ensure
+            # thread-safe execution
+            arguments = [(k, v.copy(), str(image)) for k, v in vals.items()]
+            futures = [
+                executor.submit(self.__validation_task, *arg) for arg in arguments
+            ]
+            # await results (output dropped as `vals` is updated within function)
+            for future in futures:
+                vals.update(future.result())
+
+            return CosignValidator.__apply_policy(
+                vals=vals, threshold=threshold, required=required
+            )
+        # enrich all errors with some contextual information
+        except BaseConnaisseurException as err:
+            err.update_context(**self.generate_logging_context(image, **kwargs))
```

Comment:
https://github.com/sse-secure-systems/connaisseur/blob/48088e78498c526be6254fef896c6e2b6c0af58f/connaisseur/flask_application.py#L186-L188

doesn't it make sens to add this in `flask_application.py` instead of the individual validators? then `static_validator.py` just needs to add the image in its error message.

Reply:
I was thinking along the lines of this enabling tackling the issue for better logging on each validator, but you're right that this is something that can be handled at the higher level. I'll change it